firmware/application/tone_key.cpp

```diff
@@ -114,10 +114,58 @@ std::string tone_key_string(tone_index index) {
     return tone_keys[index].first;
 }
 
-std::string tone_key_string_by_value(uint32_t value) {
-    return tone_key_string(tone_key_index_by_value(value));
+// Return string showing frequency only from specific table index
+std::string tone_key_value_string(tone_index index) {
+    if (index < 0 || (unsigned)index >= tone_keys.size())
+        return std::string("");
+    auto f = tone_keys[index].second;
+    return to_string_dec_uint((uint32_t)f) + "." + to_string_dec_uint((uint32_t)(f * 10) % 10);
+}
+
+// Return variable-length string showing CTCSS tone from tone frequency
+// Value is in 0.01 Hz units
+std::string tone_key_string_by_value(uint32_t value, size_t max_length) {
+    static uint8_t tone_display_toggle{0};
+    static uint32_t last_value;
+    tone_index idx;
+    std::string freq_str;
+
+    // If >10Hz difference between consecutive samples, it's probably noise, so ignore
+    if (abs(value - last_value) > 10 * 100) {
```

Comment:
nit: 10 * 100?

Reply:
The incoming tone frequency values from the baseband code are in 1/100 Hz increments, which I found odd (centihertz?), but without changing that I thought 10 * 100 helped to convey better that the comparison was looking for 10 Hz.  I could add a #define or something it it would help.